src/lambdas/transfer-messages/src/index.ts

```diff
@@ -0,0 +1,39 @@
+import { isError } from "shared"
+import { AwsS3Gateway } from "@bichard/s3"
+import createS3Config from "./createS3Config"
+import TransferMessagesUseCase from "./TransferMessagesUseCase"
+import type { TransferMessagesInput, TransferMessagesOptions, TransferMessagesResult } from "./types"
+
+const externalBucketGateway = new AwsS3Gateway(createS3Config())
+const internalS3BucketName = process.env.INTERNAL_INCOMING_MESSAGE_BUCKET_NAME
+
+if (!internalS3BucketName) {
+  throw new Error("INTERNAL_INCOMING_MESSAGE_BUCKET_NAME environment variable must have value")
+}
+
+const transferMessages = new TransferMessagesUseCase(externalBucketGateway, internalS3BucketName)
+
+export default async ({ numberOfObjectsToTransfer }: TransferMessagesInput): Promise<TransferMessagesResult> => {
+  if (
+    !numberOfObjectsToTransfer ||
+    (Number.isNaN(Number(numberOfObjectsToTransfer)) && numberOfObjectsToTransfer !== "all")
+  ) {
+    throw Error("Provided numberOfObjectsToTransfer is invalid")
+  }
+
+  const transferMessagesOptions: TransferMessagesOptions = {}
+
+  if (!Number.isNaN(Number(numberOfObjectsToTransfer))) {
+    transferMessagesOptions.numberOfObjectsToTransfer = parseInt(numberOfObjectsToTransfer.toString(), 10)
+  }
+
+  const result = await transferMessages.execute(transferMessagesOptions)
+
+  if (isError(result)) {
+    throw result
+  }
+
+  // eslint-disable-next-line no-console
+  console.log(result)
```

Comment:
why do we output the result here? is this to be used by logging later?

Reply:
Yes, that's right. I was thinking maybe we can use these logs in logstash to know which messages have been successfully transferred and which ones have been failed.

I'm checking if Event Bridge can also log lambda's result in cloudwatch so then we can remove this.